components/date-picker/types/DatePickerPlaceholder.cs

```diff
@@ -1,38 +1,34 @@
-
-namespace AntDesign
+﻿namespace AntDesign
 {
     internal class DatePickerPlaceholder
     {
-
-
-        public static string GetPlaceholderByType(string pickerType, CultureInfo info)
+        public static string GetPlaceholderByType(string pickerType, DatePickerLocale locale)
         {
             var placeholder = pickerType switch
             {
-                DatePickerType.Date => info.GetDateLocale().SelectDate,
-                DatePickerType.Week => info.GetDateLocale().SelectWeek,
-                DatePickerType.Month => info.GetDateLocale().SelectMonth,
-                DatePickerType.Quarter => info.GetDateLocale().SelectQuarter,
-                DatePickerType.Year => info.GetDateLocale().SelectYear,
-                DatePickerType.Time => info.GetDateLocale().SelectTime,
-                _ => info.GetDateLocale().SelectDate,
+                DatePickerType.Date => locale.Lang.DateSelect,
+                DatePickerType.Week => locale.Lang.WeekSelect,
+                DatePickerType.Month => locale.Lang.MonthSelect,
+                DatePickerType.Quarter => locale.Lang.QuarterSelect,
+                DatePickerType.Year => locale.Lang.YearSelect,
+                DatePickerType.Time => locale.Lang.TimeSelect,
+                _ => locale.Lang.DateSelect,
             };
 
             return placeholder;
         }
 
-        public static (string, string) GetRangePlaceHolderByType(string pickerType, CultureInfo info)
+        public static (string, string) GetRangePlaceHolderByType(string pickerType, DatePickerLocale locale)
         {
             var placeholder = pickerType switch
             {
-                DatePickerType.Date => (info.GetDateLocale().StartOfDate, info.GetDateLocale().EndOfDate),
-                DatePickerType.Week => (info.GetDateLocale().StartOfWeek, info.GetDateLocale().EndOfWeek),
-                DatePickerType.Month => (info.GetDateLocale().StartOfMonth, info.GetDateLocale().EndOfMonth),
-                DatePickerType.Year => (info.GetDateLocale().StartOfYear, info.GetDateLocale().EndOfYear),
-                DatePickerType.Time => (info.GetDateLocale().StartOfDate, info.GetDateLocale().EndOfDate),
-                DatePickerType.Quarter => (info.GetDateLocale().StartOfQuarter, info.GetDateLocale().EndOfQuarter),
-                _ => (info.GetDateLocale().StartOfDate, info.GetDateLocale().EndOfDate),
+                DatePickerType.Date => (locale.Lang.StartDate, locale.Lang.EndDate),
+                DatePickerType.Week => (locale.Lang.StartWeek, locale.Lang.EndWeek),
+                DatePickerType.Month => (locale.Lang.StartMonth, locale.Lang.EndMonth),
+                DatePickerType.Year => (locale.Lang.StartYear, locale.Lang.EndYear),
+                DatePickerType.Time => (locale.Lang.StartDate, locale.Lang.EndDate),
+                DatePickerType.Quarter => (locale.Lang.StartQuarter, locale.Lang.EndQuarter),
+                _ => (locale.Lang.StartDate, locale.Lang.EndDate),
```

Comment:
Why not use the existing `RangePlaceholder`,`RangeYearPlaceholder`,`RangeMonthPlaceholder` and `RangeWeekPlaceholder`?

Reply:
We may need to add `RangeQuarterPlaceholder`

Reply:
Simply because I overlooked them.

Reply:
Simply because I overlooked them.
Also, maybe it's just me, but using an array of strings for localization feels a bit weird in C# since you would have to access it afterwards in code with the index which in my opinion is rather intuitive. So far at least from the code references in VS they are used nowhere so we still can go either way.